r/vignettes/prevdocs.Rmd

```diff
@@ -0,0 +1,12 @@
+---
+title: "Current and Older Versions of this Documentation"
+output: rmarkdown::html_vignette
+---
+
+```{r, echo=FALSE}
+versions <- jsonlite::read_json("https://arrow.apache.org/docs/r/versions.json")
```

Comment:
How about reading local `r/pkgdown/assets/versions.json` instead of `https://arrow.apache.org/docs/r/versions.json`?
It seems that we don't need to use https://arrow.apache.org/docs/r/versions.json here.